### DISCUSSION TITLE
Translations through Weblate

### DISCUSSION THREAD

**Discussion by mpaglia0** [mosparo/mosparo#172] [Q&A] [answered]
Hi,
Weblate does not permit to translate because the project is in maintenance status.
I did a try some weeks ago and found the same problem.
Is this OK or Weblate translations are blocked and Mosparo needs .po files through pull requests?

Thanks,
Maurizio

**Comment by zepich** [comment]
Hi Maurizio

Thank you very much for your question.

Yes, we've put the `master` branches in Weblate into maintenance mode (yellow marked), so we do not have problems with the `develop` branch. If someone adds something in the `master` branch now that we have the next version ready, we have a problem with the translation files in the `develop` branch.

So, if you want to help translate mosparo, please use the `develop` branches in Weblate (red arrows):

![image](https://github.com/mosparo/mosparo/assets/391914/57dbd372-8c5b-4e52-a04f-87e71ab1a4aa)

Happy new year!

Kind regards,

zepich

**Comment by zepich** [comment]
Hi Maurizio

As far as I can see, you can now work with Weblate, right? Is everything okay, or do you need any assistance?

Thank you very much for your help with translating mosparo!

Kind regards,

zepich

**Reply by mpaglia0** [reply]
Ciao, @zepich 
it is the first time I use Weblate and found it uncomfortable and time consuming.
Moreover strings are duplicated several time (for example 'User') and must be translated the same.
In case of 'User' it is easy but when strings belongs to popups or descriptions this id tricky...
I usually use .po files and life is easier this way.
Antway I will proceed.
Thanks,
Maurizio

**Reply by zepich** [reply]
Hi Maurizio

Thank you very much for your help with the translations!

The duplicated strings are my 'fault'. Of course, it was not a fault initially, but after all the development, I created some duplicated strings. We've decided not to use the actual word as a source for the translation but instead use unique keys. Every string has its own key, which results in duplicated strings. I already have that on my list to try to reverse in some places to reduce the number of duplicated strings in the future (for example, 'User').

Theoretically, you could also send the YAML file with the translations directly in a pull request (without using Weblate). For this, use the English translation file as a base, translate all the strings, and create a pull request. But since the Italian translation was already started, it's impossible to get the YAML file with the missing translations (as far as I know).

Thank you very much for all your help, and have a lovely evening!

Kind regards,

zepich

**Reply by mpaglia0** [reply]
Ciao @zepich 
I do not want to find someone 'fault' but simply inform I have some difficulties with Weblate as a tool.
I find more fast, simple and effective a text editor like my Lokalize or PoEdit...
Weblate user interface is more complicated, time consuming and cannot scroll up/down easily like I do with a text editor.
Anyway I am proceeding. 

Maurizio

**Comment by mpaglia0** [comment, marked as answer]
Dear @zepich 
Italian translation should be finished!
Since I am not so skilled with Weblate please confirm all is OK.
Thank you!!
Maurizio

**Reply by zepich** [reply]
Hi @mpaglia0 

Thank you very much for all your hard work! You're amazing!

The translations look good so far. I will do an extensive review, but so far, everything is perfect and correctly translated (in terms of Weblate; unfortunately, I don't understand Italian... :) ).

Thank you very much!

I wish you a nice day!

Kind regards,

zepich
